modules/new_serverpod_auth/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,404 @@
+
+    if (useCache && userInfo != null) {
+      await session.caches.local.put(
+        cacheKey,
+        userInfo,
+        lifetime: UserProfileConfig.current.userInfoCacheLifetime,
+      );
+    }
+
+    return userInfo;
+  }
+
+  /// Updates a profiles's user name.
+  static Future<UserProfileModel> changeUserName(
+    final Session session,
+    final UuidValue authUserId,
+    final String? newUserName,
+  ) async {
+    final userProfile = await _findUserProfile(
+      session,
+      authUserId,
+    );
+
+    userProfile.userName = newUserName;
+
+    final updatedProfile = await _updateProfile(session, userProfile);
+
+    return updatedProfile.toModel();
+  }
+
+  /// Updates a profile's full name.
+  static Future<UserProfileModel> changeFullName(
+    final Session session,
+    final UuidValue authUserId,
+    final String? newFullName,
+  ) async {
+    final userProfile = await _findUserProfile(
+      session,
+      authUserId,
+    );
+
+    userProfile.fullName = newFullName;
+
+    final updatedProfile = await _updateProfile(session, userProfile);
+
+    return updatedProfile.toModel();
+  }
+
+  /// Remove the user profile for the given [authUserId].
+  ///
+  /// In case the user did not have a profile, nothing is changed.
+  static Future<void> deleteProfileForUser(
+    final Session session,
+    final UuidValue authUserId,
+  ) async {
+    await UserProfile.db.deleteWhere(
+      session,
+      where: (final t) => t.authUserId.equals(authUserId),
+    );
+
+    await _invalidateCacheForUser(session, authUserId);
+  }
+
+// #region Profile images
+
+  static Future<UserProfileImage> _createImageFromUrl(
+    final Session session,
+    final UuidValue authUserId,
+    final Uri url,
+  ) async {
+    final result = await http.get(url);
+    final bytes = result.bodyBytes;
+
+    return _createImageFromBytes(session, authUserId, bytes);
+  }
+
+  static Future<UserProfileImage> _createImageFromBytes(
+    final Session session,
+    final UuidValue authUserId,
+    final Uint8List imageBytes,
+  ) async {
+    final reEncodedImageBytes = await Isolate.run(() async {
+      var image = decodeImage(imageBytes)!;
+
+      final imageSize = UserProfileConfig.current.userImageSize;
+      if (image.width != imageSize || image.height != imageSize) {
+        image = copyResizeCropSquare(
+          image,
+          size: imageSize,
+          interpolation: Interpolation.average,
+        );
+      }
+
+      return _encodeImage(image);
+    });
+
+    final userProfile = await _findUserProfile(session, authUserId);
+
+    // Find the latest version of the user image if any.
+    final oldImageRef = await UserProfileImage.db.findFirstRow(
+      session,
+      where: (final t) => t.userProfileId.equals(userProfile.id!),
+      orderBy: (final t) => t.version,
+      orderDescending: true,
+    );
+
+    // Add one to the version number or create a new version 1.
+    final version = (oldImageRef?.version ?? 0) + 1;
```

Comment:
Do we think the version is actually relevant here, or couldn’t we just give the image a unique name (random part or timestamp).

It’s then referenced via the user profile anyway (and not looked up by most recent version).

Reply:
Decision:
- We do not need the numeric ID, random identifier / UUID is fine.

Reply:
Switched to a short random part for every file's name (also still scoped to the user id).